pkg/okteto/registry_credentials.go

```diff
@@ -5,15 +5,56 @@ import (
 	"fmt"
 	"net/url"
 	"strings"
+	"sync"
 	"time"
 
 	dockertypes "github.com/docker/cli/cli/config/types"
 	oktetoLog "github.com/okteto/okteto/pkg/log"
 )
 
+var globalRegistryCache registryCache
+
+type registryCache struct {
+	cache map[string][2]string
```

Comment:
NIT: Could we create a custom type instead of having `[2]string`. I think it'd be more readable if we have a custom type `registryCacheCredentials` with User and pass. WDYT?